sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c

```diff
@@ -417,7 +417,7 @@ static size_t _iphc_ipv6_decode(const uint8_t *iphc_hdr,
                 ipv6_hdr->dst.u8[1] = iphc_hdr[payload_offset++];
                 ipv6_hdr->dst.u8[2] = iphc_hdr[payload_offset++];
                 ipv6_hdr->dst.u8[3] = ctx->prefix_len;
-                ipv6_addr_init_prefix((ipv6_addr_t *)ipv6_hdr->dst.u8 + 4,
+                ipv6_addr_init_prefix((ipv6_addr_t *)(ipv6_hdr->dst.u8 + 4),
                                       &ctx->prefix, ctx->prefix_len);
                 memcpy(ipv6_hdr->dst.u8 + 12, iphc_hdr + payload_offset + 2, 4);
```

Comment:
This expresses the following paragraph in [RFC 6282]

>       If M=1 and DAC=1:
> 
>          00:  48 bits.  This format is designed to match Unicast-Prefix-
>             based IPv6 Multicast Addresses as defined in [RFC3306] and
>             [RFC3956].  The multicast address takes the form ffXX:XXLL:
>             PPPP:PPPP:PPPP:PPPP:XXXX:XXXX. where the X are the nibbles
>             that are carried in-line, in the order in which they appear
>             in this format.  P denotes nibbles used to encode the prefix
>             itself.  L denotes nibbles used to encode the prefix length.
>             The prefix information P and L is taken from the specified
>             context.

So :heavy_check_mark: 

[RFC 6282]: https://tools.ietf.org/html/rfc6282#section-3.1.1